statwrap/fpp.py

```diff
@@ -112,10 +112,16 @@ def scatter_plot(x, y, xlim=None, ylim=None,
     y = np.squeeze(np.array(y))
     ax.scatter(x, y, **kwargs)
 
-    if regression_line:
-        m, b = np.polyfit(x, y, 1)  # Calculating the slope (m) and intercept (b) of the regression line
+    m, b = np.polyfit(x, y, 1)  # Calculating the slope (m) and intercept (b) of the regression line
+    
+    if regression_line:    
         ax.plot(x, m*x + b, color='gray')  # Plotting the regression line
 
+    if regression_equation: 
+        equation_text = f'y = {m:.2f}x + {b:.2f}'  # Add regression line equation to plot
+        ax.text(0.5 , 1, equation_text, transform=ax.transAxes, fontsize=10,
```

Comment:
are you missing a pad param in `ax.set_title` further below? You should write this so the pad is `None` unless regression_equation is True.